File: Tutorials/Codes/Introduction_to_Robotics_Course/Chapter7/Lesson4/Chapter7_Lesson4.py
import cv2

# RGB camera (index 0)
cap = cv2.VideoCapture(0)
if not cap.isOpened():
    raise RuntimeError("Cannot open camera")

while True:
    ok, frame = cap.read()
    if not ok:
        break

    # frame is BGR in OpenCV; convert to RGB if needed
    rgb = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
    cv2.imshow("RGB stream", frame)

    if cv2.waitKey(1) & 0xFF == ord('q'):
        break

cap.release()
cv2.destroyAllWindows()

File: Tutorials/Codes/Introduction_to_Robotics_Course/Chapter7/Lesson4/Chapter7_Lesson4.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

int main() {
    cv::VideoCapture cap(0);
    if (!cap.isOpened()) {
        std::cerr << "Cannot open camera\n";
        return -1;
    }

    cv::Mat frame;
    while (true) {
        cap >> frame;
        if (frame.empty()) break;

        cv::imshow("RGB stream", frame);
        if (cv::waitKey(1) == 'q') break;
    }
    return 0;
}
